### ISSUE TITLE
RuntimeError: maximum recursion depth exceeded in cmp

### ISSUE THREAD

**Issue by shimizukawa** [sphinx-doc/sphinx#1510]
I have opened a new ticket since #1347 is marked as solved.

Was this  fixed or I hit a different issue?

I am using 1.2.2 and I got

```
#!python

File "/home/adi/chevah/venv/lib/python2.7/site-packages/sphinx/themes/basic/page.html", line 10, in top-level template code
    {&#37;- extends "layout.html" &#37;}
  File "/home/adi/chevah/venv/lib/python2.7/site-packages/sftpplus/website/sphinx/integrated/layout.html", line 1, in top-level template code
    {&#37; extends "!layout.html" &#37;}
  File "/home/adi/chevah/venv/lib/python2.7/site-packages/sftpplus/website/sphinx/integrated/layout.html", line 1, in top-level template code
    {&#37; extends "!layout.html" &#37;}

...

  File "/home/adi/chevah/venv/lib/python2.7/site-packages/jinja2/utils.py", line 339, in get
    return self[key]
  File "/home/adi/chevah/venv/lib/python2.7/site-packages/jinja2/utils.py", line 389, in __getitem__
    if self._queue[-1] != key:
RuntimeError: maximum recursion depth exceeded in cmp

```

My inheritance is like this:

'INTEGRATED_CUSTOM theme' -> 'OTHER_CUSTOM theme' -> basic

So I have a 'OTHER_CUSTOM' theme which inherits from basic... but defines its own layout.html without extending the layout from basic.

Then I have 'INTEGRATED_CUSTOM theme, which inherits from OTHER_CUSTOM and extend the layout.html

Thanks!

---
- Bitbucket: https://bitbucket.org/birkenfeld/sphinx/issue/1510
- Originally reported by: Adi Roiban
- Originally created at: 2014-07-13T11:56:08.752

**Comment by shimizukawa** [comment]
_From Takayuki Shimizukawa on 2014-07-13 13:29:51+00:00_

small sample to reproduce the issue.

**Comment by shimizukawa** [comment]
_From Takayuki Shimizukawa on 2014-07-13 13:34:41+00:00_

I think `INTEGRATED_CUSTOM theme` is using `{%- extends "!layout.html" %}` but it is not correct. It should be `{%- extends "OTHER_CUSTOM/layout.html" %}` instead.

Exclamation prefixing should be used by a user custom template. Please change your INTEGRATED_CUSTOM theme `extends` line.

Thanks.

**Comment by shimizukawa** [comment]
_From Adi Roiban on 2014-07-13 14:06:42+00:00_

Thanks for the help...

yes, in INTEGRATED_CUSTOM I was only using `{%- extends "!layout.html" %}` as I was thinking that this is the hook for the inherited theme.

I read again the documentation and now I saw the example... I have no idea how I ended up with the exclamation mark.

in INTEGRATED_CUSTOM theme.conf file ,  I already had the inherit configuration so I thought there is no need to repeat it in the templates.

This ticket is invalid.
Sorry for the noise!
